### PR TITLE
Git commit loadtest

### PR DESCRIPTION
**Pull request by sukantoraymond**

### CODE REVIEW

File: cmd/nodecmd/loadtest.go [hunk 1]
```diff
@@ -404,10 +411,32 @@ func createClusterYAMLFile(clusterName, subnetID, chainID string, separateHost *
 	return enc.Encode(clusterInfoYAML)
 }
 
+func getGitCommit(gitRepoURL string) string {
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:414]: move to utils/common
add unittests

File: cmd/nodecmd/loadtest.go [hunk 2]
```diff
@@ -404,10 +411,32 @@ func createClusterYAMLFile(clusterName, subnetID, chainID string, separateHost *
...
+	if strings.Contains(gitRepoURL, "/commit/") {
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:415]: add check that gitRepoUrl is a valid url
add check len(splitURL)>=1

File: cmd/nodecmd/loadtest.go [hunk 3]
```diff
@@ -404,10 +411,32 @@ func createClusterYAMLFile(clusterName, subnetID, chainID string, separateHost *
...
+		splitURL := strings.Split(gitRepoURL, "/")
+		commitID := splitURL[len(splitURL)-1]
+		return commitID
+	}
+	return ""
+}
+
+func getRepoFromCommitURL(gitRepoURL string) string {
+	splitURL := strings.Split(gitRepoURL, "/")
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:424]: we need to check if url is valid, we need checks for len of the slice to make sure it doesn't crash

File: cmd/nodecmd/loadtest.go [hunk 4]
```diff
@@ -404,10 +411,32 @@ func createClusterYAMLFile(clusterName, subnetID, chainID string, separateHost *
...
+	splitURLWOCommit := splitURL[:len(splitURL)-2]
+	repoDirName = splitURLWOCommit[len(splitURLWOCommit)-1]
+	gitRepo := strings.Join(splitURLWOCommit[:], "/")
+	gitRepo += ".git"
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:428]: nit: we have `const gitExtension = ".git"` maybe we should use it there too so we reuse const

File: cmd/nodecmd/loadtest.go [hunk 5]
```diff
@@ -418,6 +447,10 @@ func GetLoadTestScript(app *application.Avalanche) error {
 		if err != nil {
 			return err
 		}
+		loadTestRepoCommit = getGitCommit(loadTestRepoURL)
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:450]: why do we do it twice ? 
maybe move it up from the if{} statement

File: pkg/ssh/shell/buildLoadTest.sh [hunk 6]
```diff
@@ -2,6 +2,9 @@
 # get most updated load test repo
 echo "getting load test repo ..."
 git -C {{ .LoadTestRepoDir }} pull || git clone {{ .LoadTestRepo }}
+{{if .CheckoutCommit }}
+cd {{ .RepoDirName}}; git checkout {{ .LoadTestGitCommit}}
```
Comment (arturrez) [inline pkg/ssh/shell/buildLoadTest.sh:6]: nit: use `pushd` `popd` so `cwd` states the same after the operation

File: pkg/ssh/ssh.go [hunk 7]
```diff
@@ -433,7 +436,7 @@ func RunSSHSetupBuildEnv(host *models.Host) error {
 	)
 }
 
-func RunSSHBuildLoadTest(host *models.Host, loadTestRepo, loadTestPath string) error {
+func RunSSHBuildLoadTest(host *models.Host, loadTestRepo, loadTestPath, loadTestGitCommit, repoDirName string, checkoutCommit bool) error {
 	loadTestRepoPaths := strings.Split(loadTestRepo, "/")
 	// remove .git
 	loadTestRepoDir := strings.Split(loadTestRepoPaths[len(loadTestRepoPaths)-1], ".")
```
Comment (arturrez) [inline pkg/ssh/ssh.go:442]: nit: check for len of slice to make sure it doesn't crash

File: pkg/ssh/shell/buildLoadTest.sh [hunk 8]
```diff
@@ -2,6 +2,9 @@
 # get most updated load test repo
 echo "getting load test repo ..."
 git -C {{ .LoadTestRepoDir }} pull || git clone {{ .LoadTestRepo }}
+{{if .CheckoutCommit }}
```
Comment (arturrez) [inline pkg/ssh/shell/buildLoadTest.sh:5]: do we need `.CheckoutCommit`  if we can check `.LoadTestGitCommit` is not empty ?

File: cmd/nodecmd/loadtest.go [hunk 9]
```diff
@@ -404,10 +411,32 @@ func createClusterYAMLFile(clusterName, subnetID, chainID string, separateHost *
...
+		splitURL := strings.Split(gitRepoURL, "/")
+		commitID := splitURL[len(splitURL)-1]
+		return commitID
+	}
+	return ""
+}
+
+func getRepoFromCommitURL(gitRepoURL string) string {
+	splitURL := strings.Split(gitRepoURL, "/")
+	splitURLWOCommit := splitURL[:len(splitURL)-2]
```
Comment (arturrez) [inline cmd/nodecmd/loadtest.go:425]: add comment with url example so it's easy to read this logic

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**